spark/src/main/java/org/opensearch/sql/spark/client/EMRServerlessClientFactoryImpl.java

```diff
@@ -0,0 +1,66 @@
+package org.opensearch.sql.spark.client;
+
+import static org.opensearch.sql.common.setting.Settings.Key.SPARK_EXECUTION_ENGINE_CONFIG;
+
+import com.amazonaws.auth.DefaultAWSCredentialsProviderChain;
+import com.amazonaws.services.emrserverless.AWSEMRServerless;
+import com.amazonaws.services.emrserverless.AWSEMRServerlessClientBuilder;
+import java.security.AccessController;
+import java.security.PrivilegedAction;
+import lombok.RequiredArgsConstructor;
+import org.opensearch.sql.spark.config.SparkExecutionEngineConfig;
+import org.opensearch.sql.spark.config.SparkExecutionEngineConfigSupplier;
+
+/** Implementation of {@link EMRServerlessClientFactory}. */
+@RequiredArgsConstructor
+public class EMRServerlessClientFactoryImpl implements EMRServerlessClientFactory {
+
+  private final SparkExecutionEngineConfigSupplier sparkExecutionEngineConfigSupplier;
+  private EMRServerlessClient emrServerlessClient;
+  private String region;
+
+  /**
+   * Gets an instance of {@link EMRServerlessClient}.
+   *
+   * @return An {@link EMRServerlessClient} instance.
+   */
+  @Override
+  public EMRServerlessClient getClient() {
+    SparkExecutionEngineConfig sparkExecutionEngineConfig =
+        this.sparkExecutionEngineConfigSupplier.getSparkExecutionEngineConfig();
+    validateSparkExecutionEngineConfig(sparkExecutionEngineConfig);
+    if (isNewClientCreationRequired(sparkExecutionEngineConfig.getRegion())) {
+      region = sparkExecutionEngineConfig.getRegion();
+      this.emrServerlessClient = createEMRServerlessClient(this.region);
```

Comment:
If client changed, how to get job status of in previous region?
do we need to shutdown underlying EMRClient?

Reply:
usecase I am targeting is customer initially didn't configure spark execution engine cluster setting(or configured wrong region) and once the customer configures we create a new client. Also, Lazy client creation when async api is first invoked.


If customer changes region in between, then it will result in errors for old jobs. Customer needs to be careful enough.[Do you see this as issue?]

Reply:
understood, could we add notes in our doc to alert customer?

Reply:
Ok.

Reply:
Addressing in this PR: https://github.com/opensearch-project/sql/pull/2493